### PR TITLE
sca2d: init at 0.2.0

### PR DESCRIPTION
**Pull request by traxys**
###### Description of changes

[sca2d](https://gitlab.com/bath_open_instrumentation_group/sca2d) is a static analyzer for [OpenSCAD](https://openscad.org/)

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Alizter** [comment]
Result of `nixpkgs-review pr 223097` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>sca2d</li>
    <li>sca2d.dist</li>
  </ul>
</details>

**Review by Alizter — APPROVED** [review]

File: pkgs/development/tools/sca2d/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{ lib
+, python3
+, fetchFromGitLab
+, fetchFromGitHub
+}:
+let
+  lark010 = python3.pkgs.lark.overrideAttrs (old: rec {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/sca2d/default.nix:7]: This just be fine via python package overwrites otherwise it is not reliable

Comment (traxys) [reply pkgs/development/tools/sca2d/default.nix:7]: Sorry I didn't understand your point ? I took gdtoolkit (https://github.com/NixOS/nixpkgs/blob/798fd6a3ffb7fae359f6e02a379bfb188d54fcad/pkgs/development/tools/gdtoolkit/default.nix#L7-L17) as example

Comment (GaetanLepage) [reply pkgs/development/tools/sca2d/default.nix:7]: @SuperSandro2000 are you taking about using [`overridePythonAttrs`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/python-packages-base.nix#L16-L34) ?

Comment (SuperSandro2000) [reply pkgs/development/tools/sca2d/default.nix:7]: yes, https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#overriding-python-packages-overriding-python-packages

Comment (traxys) [reply pkgs/development/tools/sca2d/default.nix]: @SuperSandro2000 Done!

File: pkgs/development/tools/sca2d/default.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          };
+
+          disabledTestPaths = [ "tests/test_nearley/test_nearley.py" ];
+        });
+      };
+    in
+    python3.override {
+      inherit packageOverrides;
+      self = python;
+    };
```
Comment (SuperSandro2000) [inline pkgs/development/tools/sca2d/default.nix:28]:
```suggestion
  python = python3.override {
    packageOverrides = self: super: {
      lark010 = super.lark.overridePythonAttrs (old: rec {
        version = "0.10.0";

        src = fetchFromGitHub {
          owner = "lark-parser";
          repo = "lark";
          rev = "refs/tags/${version}";
          sha256 = "sha256-ctdPPKPSD4weidyhyj7RCV89baIhmuxucF3/Ojx1Efo=";
        };

        disabledTestPaths = [ "tests/test_nearley/test_nearley.py" ];
      });
    };
    self = python;
  };
```

Comment (traxys) [reply pkgs/development/tools/sca2d/default.nix]: Fixed!

File: pkgs/development/tools/sca2d/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{ lib
+, python3
+, fetchFromGitLab
+, fetchFromGitHub
+,
```
Comment (SuperSandro2000) [inline pkgs/development/tools/sca2d/default.nix:5]:
```suggestion
```

**Pull request merged**